dapps/marketplace/src/pages/identity/EmailAttestationModal.js

```diff
@@ -34,12 +34,17 @@ class EmailAttestationModal extends Component {
           } else if (this.props.onClose) {
             this.props.onClose()
           }
+
+          this.setState({
+            shouldClose: false
+          })
         }}
         lightMode={true}
         skipAnimateOnExit={this.props.skipAnimateOnExit}
       >
         <EmailAttestation
           wallet={this.props.wallet}
+          close={() => this.setState({ shouldClose: true })}
```

Comment:
@sparrowDom Are you change you intended to change this line?

If I understand this correctly, I was able to cancel before it sent me a verification code. However, in the "Verify Code" screen, I cannot cancel. But for that to be fixed, shouldn't you [update this line](https://github.com/OriginProtocol/origin/blob/ac9056b6d50e10780e049ab1d9f275c15b5bab22/dapps/marketplace/src/pages/identity/EmailAttestation.js#L367)?

Reply:
Yeah you are right, I have miss-diagnosed the problem. The modal wouldn't close when clicking on cancel only the second time you click it. It is corrected now.

Reply:
Thank you @sparrowDom